### ISSUE TITLE
[meta] panicked at 'called `Result::unwrap()` on an `Err` value: Code: 2407, displayText = targeting version 1, current version 2.

### ISSUE THREAD

**Issue by BohuTANG** [datafuselabs/databend#2589]
**Summary**

Failed at https://github.com/datafuselabs/databend/runs/4070390168?check_suite_focus=true

Supplementary info:

- A first guess is that `MetaCached` cached table info thus no updated table can be re-fetched in a CAS-loop. This is fixed in #2603 . Keep watching other CI to see if this exception disspears.
- Another issue is the `TableVersionMissMatch` error is not dealt with. The following TODO is meant to get this done:

https://github.com/datafuselabs/databend/blob/088f8bbb95d2a7c04cd0fe67df795765cd23ce3f/query/src/datasources/table/fuse/table_do_append.rs#L70-L96

```
test servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table has been running for over 60 seconds
test servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table ... FAILED

failures:

---- servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table stdout ----
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Code: 2407, displayText = targeting version 1, current version 2.

   0: common_exception::exception::ErrorCode::TableVersionMissMatch
             at /home/runner/work/databend/databend/common/exception/src/exception.rs:110:77
   1: common_meta_embedded::meta_api_impl::<impl common_meta_api::meta_api::MetaApi for common_meta_embedded::meta_embedded::MetaEmbedded>::upsert_table_option::{{closure}}
             at /home/runner/work/databend/databend/common/meta/embedded/src/meta_api_impl.rs:266:24
   2: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
   3: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/future.rs:119:9
   4: <databend_query::catalogs::backends::impls::meta_cached::MetaCached as common_meta_api::meta_api::MetaApi>::upsert_table_option::{{closure}}
             at src/catalogs/backends/impls/meta_cached.rs:113:9
   5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
   6: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/future.rs:119:9
   7: <databend_query::catalogs::backends::impls::meta_sync::MetaSync as databend_query::catalogs::backends::backend::MetaApiSync>::upsert_table_option::{{closure}}
             at src/catalogs/backends/impls/meta_sync.rs:117:13
   8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
   9: <T as common_base::runtime::BlockingWait>::wait_in::{{closure}}
             at /home/runner/work/databend/databend/common/base/src/runtime.rs:130:21
  10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
  11: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/core.rs:161:17
  12: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/loom/std/unsafe_cell.rs:14:9
  13: tokio::runtime::task::core::CoreStage<T>::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/core.rs:151:13
  14: tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:461:19
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
  15: std::panicking::try::do_call
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
  16: std::panic::catch_unwind
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
  17: tokio::runtime::task::harness::poll_future
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:449:18
  18: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:98:27
  19: tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:53:15
  20: tokio::runtime::task::raw::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/raw.rs:113:5
  21: tokio::runtime::task::raw::RawTask::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/raw.rs:70:18
  22: tokio::runtime::task::LocalNotified<S>::run
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/mod.rs:343:9
  23: tokio::runtime::thread_pool::worker::Context::run_task::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:420:13
  24: tokio::coop::with_budget::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/coop.rs:106:9
  25: std::thread::local::LocalKey<T>::try_with
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/local.rs:399:16
  26: std::thread::local::LocalKey<T>::with
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/local.rs:375:9
  27: tokio::coop::with_budget
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/coop.rs:99:5
      tokio::coop::budget
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/coop.rs:76:5
      tokio::runtime::thread_pool::worker::Context::run_task
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:419:9
  28: tokio::runtime::thread_pool::worker::Context::run
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:386:24
  29: tokio::runtime::thread_pool::worker::run::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:371:17
      tokio::macros::scoped_tls::ScopedKey<T>::set
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/macros/scoped_tls.rs:61:9
  30: tokio::runtime::thread_pool::worker::run
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:368:5
  31: tokio::runtime::thread_pool::worker::Launch::launch::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:347:45
      <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/blocking/task.rs:42:21
  32: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/core.rs:161:17
  33: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/loom/std/unsafe_cell.rs:14:9
  34: tokio::runtime::task::core::CoreStage<T>::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/core.rs:151:13
  35: tokio::runtime::task::harness::poll_future::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:461:19
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
  36: std::panicking::try::do_call
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
  37: std::panic::catch_unwind
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
  38: tokio::runtime::task::harness::poll_future
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:449:18
  39: tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:98:27
  40: tokio::runtime::task::harness::Harness<T,S>::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:53:15
  41: tokio::runtime::task::raw::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/raw.rs:113:5
  42: tokio::runtime::task::raw::RawTask::poll
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/raw.rs:70:18
  43: tokio::runtime::task::UnownedTask<S>::run
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/mod.rs:379:9
  44: tokio::runtime::blocking::pool::Inner::run
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/blocking/pool.rs:265:17
  45: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/blocking/pool.rs:245:17
  46: std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:125:18
  47: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/mod.rs:481:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
  48: std::panicking::try::do_call
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
  49: std::panic::catch_unwind
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
  50: std::thread::Builder::spawn_unchecked::{{closure}}
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/mod.rs:480:30
  51: core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/ops/function.rs:227:5
  52: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys/unix/thread.rs:106:17
  53: start_thread
  54: clone
', query/src/servers/clickhouse/interactive_worker_base.rs:137:41
stack backtrace:
   0:     0x55c781678e7c - std::backtrace_rs::backtrace::libunwind::trace::h2ab374bc2a3b7023
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x55c781678e7c - std::backtrace_rs::backtrace::trace_unsynchronized::h128cb5178b04dc46
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55c781678e7c - std::sys_common::backtrace::_print_fmt::h5344f9eefca2041f
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55c781678e7c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h213003bc5c7acf04
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x55c7816d58dc - core::fmt::write::h78bf85fc3e93663f
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/fmt/mod.rs:1125:17
   5:     0x55c781669f05 - std::io::Write::write_fmt::hf821662cc642698f
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/io/mod.rs:1667:15
   6:     0x55c78167c3e0 - std::sys_common::backtrace::_print::hf706674f77848203
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x55c78167c3e0 - std::sys_common::backtrace::print::hf0b6c7a88804ec56
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x55c78167c3e0 - std::panicking::default_hook::{{closure}}::h2dde766cd83a333a
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:210:50
   9:     0x55c78167bfd2 - std::panicking::default_hook::h501e3b2e134eb149
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:224:9
  10:     0x55c78167cbd4 - std::panicking::rust_panic_with_hook::hc09e869c4cf00885
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:624:17
  11:     0x55c78167c6b0 - std::panicking::begin_panic_handler::{{closure}}::hc2c6d70142458fc8
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:521:13
  12:     0x55c781679324 - std::sys_common::backtrace::__rust_end_short_backtrace::had58f7c459a1cd6e
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x55c78167c619 - rust_begin_unwind
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:517:5
  14:     0x55c770c62f61 - core::panicking::panic_fmt::hf443e5eeb6cc9eab
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panicking.rs:96:14
  15:     0x55c770c63053 - core::result::unwrap_failed::h885d3f7beb571353
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/result.rs:1617:5
  16:     0x55c77266ee6c - core::result::Result<T,E>::unwrap::hb0b36a772a6f77cf
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/result.rs:1299:23
  17:     0x55c77266ee6c - databend_query::servers::clickhouse::interactive_worker_base::InteractiveWorkerBase::process_insert_query::{{closure}}::{{closure}}::h8ad52a803157d260
                               at /home/runner/work/databend/databend/query/src/servers/clickhouse/interactive_worker_base.rs:137:13
  18:     0x55c7723b9f0b - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h5768a7dd11c1be78
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
  19:     0x55c771ca33f8 - tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}::h21a771740578cce8
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/core.rs:161:17
  20:     0x55c771ab240b - tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut::hc3f0b5e406fa6a6b
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/loom/std/unsafe_cell.rs:14:9
  21:     0x55c771c9fa29 - tokio::runtime::task::core::CoreStage<T>::poll::hca97c564e2baa0f4
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/core.rs:151:13
  22:     0x55c770d582c6 - tokio::runtime::task::harness::poll_future::{{closure}}::hb1bc8e5bdb02185d
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:461:19
  23:     0x55c770d582c6 - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h04d2f0bc86b2682a
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
  24:     0x55c771248e9e - std::panicking::try::do_call::h6a45ca4c17ba1c57
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
  25:     0x55c771248e9e - std::panicking::try::h96ecb51a9fcf1838
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
  26:     0x55c7711925cc - std::panic::catch_unwind::h4343d075315e4394
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
  27:     0x55c771cfb30b - tokio::runtime::task::harness::poll_future::h151d312237def10e
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:449:18
  28:     0x55c771d17cfd - tokio::runtime::task::harness::Harness<T,S>::poll_inner::h771dcb7912ca6842
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:98:27
  29:     0x55c771d85065 - tokio::runtime::task::harness::Harness<T,S>::poll::hcca90e37683f1585
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:53:15
  30:     0x55c771c41572 - tokio::runtime::task::raw::poll::h43f00bfe1f14b380
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/raw.rs:113:5
  31:     0x55c780f66dd1 - tokio::runtime::task::raw::RawTask::poll::hcfbb4dccd23e8828
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/raw.rs:70:18
  32:     0x55c780f72c03 - tokio::runtime::task::LocalNotified<S>::run::hbbe8d468874cdfc8
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/mod.rs:343:9
  33:     0x55c780fa3ef5 - tokio::runtime::thread_pool::worker::Context::run_task::{{closure}}::hc0197c64e4117906
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:420:13
  34:     0x55c780eb0a39 - tokio::coop::with_budget::{{closure}}::h9eb48da1d5655e3b
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/coop.rs:106:9
  35:     0x55c780d30639 - std::thread::local::LocalKey<T>::try_with::h017c1e54e6c87a3b
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/local.rs:399:16
  36:     0x55c780d2de83 - std::thread::local::LocalKey<T>::with::h369f6d200f436519
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/local.rs:375:9
  37:     0x55c780fa3d94 - tokio::coop::with_budget::h4e1ecef04b65c341
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/coop.rs:99:5
  38:     0x55c780fa3d94 - tokio::coop::budget::h407adec509c558eb
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/coop.rs:76:5
  39:     0x55c780fa3d94 - tokio::runtime::thread_pool::worker::Context::run_task::h0cc95f17eb55bd07
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:419:9
  40:     0x55c780fa2e2c - tokio::runtime::thread_pool::worker::Context::run::hf8490164d0b6715e
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:386:24
  41:     0x55c780e9a9e2 - tokio::runtime::thread_pool::worker::run::{{closure}}::he91a7eddc8ef889b
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:371:17
  42:     0x55c780e9a9e2 - tokio::macros::scoped_tls::ScopedKey<T>::set::h32dd3e0af24e19f4
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/macros/scoped_tls.rs:61:9
  43:     0x55c780fa2930 - tokio::runtime::thread_pool::worker::run::h505d2934ae43bd63
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:368:5
  44:     0x55c780f89a84 - tokio::runtime::thread_pool::worker::Launch::launch::{{closure}}::h1ddd879e4417869c
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/worker.rs:347:45
  45:     0x55c780f89a84 - <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll::hb77958483fd02cf7
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/blocking/task.rs:42:21
  46:     0x55c780f2c519 - tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}::h733b13b0b9153cac
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/core.rs:161:17
  47:     0x55c780ea37db - tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut::h06e77802e7babc92
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/loom/std/unsafe_cell.rs:14:9
  48:     0x55c780f29bda - tokio::runtime::task::core::CoreStage<T>::poll::h7d8bfd7c604ea7f6
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/core.rs:151:13
  49:     0x55c780d0cf6a - tokio::runtime::task::harness::poll_future::{{closure}}::h992a4d8d9ce70ec6
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:461:19
  50:     0x55c780d0cf6a - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h6c5a1f8672d56297
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
  51:     0x55c780d4b9d2 - std::panicking::try::do_call::h00426213e34ba778
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
  52:     0x55c780d4b9d2 - std::panicking::try::h75fce8b0a37114d9
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
  53:     0x55c780d2c2dd - std::panic::catch_unwind::h8bf3da14c6748d46
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
  54:     0x55c780f5c1dc - tokio::runtime::task::harness::poll_future::hfdf963345db3ab99
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:449:18
  55:     0x55c780f39f5f - tokio::runtime::task::harness::Harness<T,S>::poll_inner::h0f8a464eb07efb44
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:98:27
  56:     0x55c780f38e96 - tokio::runtime::task::harness::Harness<T,S>::poll::ha58317b0395f089f
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/harness.rs:53:15
  57:     0x55c780f673f4 - tokio::runtime::task::raw::poll::hb692ec9b71b8e589
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/raw.rs:113:5
  58:     0x55c780f66dd1 - tokio::runtime::task::raw::RawTask::poll::hcfbb4dccd23e8828
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/raw.rs:70:18
  59:     0x55c780f72d73 - tokio::runtime::task::UnownedTask<S>::run::h58a80cccd1d6d53d
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/task/mod.rs:379:9
  60:     0x55c780f84962 - tokio::runtime::blocking::pool::Inner::run::h99ed791cbdb5cc34
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/blocking/pool.rs:265:17
  61:     0x55c780f8318c - tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}::h4fe9639ba765ea8a
                               at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/blocking/pool.rs:245:17
  62:     0x55c780d1955e - std::sys_common::backtrace::__rust_begin_short_backtrace::h5c06c01a97f3c892
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:125:18
  63:     0x55c780d0dafc - std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}::h52940ee31af08d92
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/mod.rs:481:17
  64:     0x55c780d0dafc - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h9a52ef1725d2b50f
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
  65:     0x55c780d437a5 - std::panicking::try::do_call::h3695e10cbc746d5b
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
  66:     0x55c780d437a5 - std::panicking::try::h33e2d57e45b2da3b
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
  67:     0x55c780d2c473 - std::panic::catch_unwind::h9021ecc9fd236f82
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
  68:     0x55c780d3e641 - std::thread::Builder::spawn_unchecked::{{closure}}::h9bc5bc4c6985082f
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/mod.rs:480:30
  69:     0x55c780d8c3b1 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hf6d264bcb5b48448
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/ops/function.rs:227:5
  70:     0x55c781689243 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h59eef3b9c8a82350
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
  71:     0x55c781689243 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hb5bbe017c347469c
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
  72:     0x55c781689243 - std::sys::unix::thread::Thread::new::thread_start::h62931528f61e35f5
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys/unix/thread.rs:106:17
  73:     0x7f8a3294a609 - start_thread
  74:     0x7f8a326f4293 - clone
  75:                0x0 - <unknown>
Error: Code: 1000, displayText = Error insert table: Driver(Timeout).

   0: common_exception::exception::ErrorCode::UnknownException
             at /home/runner/work/databend/databend/common/exception/src/exception.rs:110:77
   1: databend_query::servers::clickhouse::clickhouse_handler_test::insert::{{closure}}
             at src/servers/clickhouse/clickhouse_handler_test.rs:192:27
   2: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
   3: databend_query::servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table::{{closure}}
             at src/servers/clickhouse/clickhouse_handler_test.rs:95:5
   4: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/future/mod.rs:80:19
   5: tokio::park::thread::CachedParkThread::block_on::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/park/thread.rs:263:54
      tokio::coop::with_budget::{{closure}}
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/coop.rs:106:9
   6: std::thread::local::LocalKey<T>::try_with
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/local.rs:399:16
   7: std::thread::local::LocalKey<T>::with
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/local.rs:375:9
   8: tokio::coop::with_budget
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/coop.rs:99:5
      tokio::coop::budget
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/coop.rs:76:5
      tokio::park::thread::CachedParkThread::block_on
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/park/thread.rs:263:31
   9: tokio::runtime::enter::Enter::block_on
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/enter.rs:151:13
  10: tokio::runtime::thread_pool::ThreadPool::block_on
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/thread_pool/mod.rs:77:9
  11: tokio::runtime::Runtime::block_on
             at /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.13.0/src/runtime/mod.rs:463:43
  12: databend_query::servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table
             at src/servers/clickhouse/clickhouse_handler_test.rs:101:5
  13: databend_query::servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table::{{closure}}
             at src/servers/clickhouse/clickhouse_handler_test.rs:71:7
      core::ops::function::FnOnce::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/ops/function.rs:227:5
  14: core::ops::function::FnOnce::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/ops/function.rs:227:5
      test::__rust_begin_short_backtrace
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:578:5
  15: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
      std::panic::catch_unwind
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
      test::run_test_in_process
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:601:18
      test::run_test::run_test_inner::{{closure}}
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:493:39
  16: test::run_test::run_test_inner::{{closure}}
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:520:37
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:125:18
  17: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/mod.rs:481:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
      std::panicking::try
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
      std::panic::catch_unwind
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
      std::thread::Builder::spawn_unchecked::{{closure}}
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/mod.rs:480:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/ops/function.rs:227:5
  18: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys/unix/thread.rs:106:17
  19: start_thread
  20: clone

thread 'servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:194:5
stack backtrace:
   0:     0x55c781678e7c - std::backtrace_rs::backtrace::libunwind::trace::h2ab374bc2a3b7023
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
   1:     0x55c781678e7c - std::backtrace_rs::backtrace::trace_unsynchronized::h128cb5178b04dc46
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55c781678e7c - std::sys_common::backtrace::_print_fmt::h5344f9eefca2041f
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55c781678e7c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h213003bc5c7acf04
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x55c7816d58dc - core::fmt::write::h78bf85fc3e93663f
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/fmt/mod.rs:1125:17
   5:     0x55c781669f05 - std::io::Write::write_fmt::hf821662cc642698f
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/io/mod.rs:1667:15
   6:     0x55c78167c3e0 - std::sys_common::backtrace::_print::hf706674f77848203
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x55c78167c3e0 - std::sys_common::backtrace::print::hf0b6c7a88804ec56
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x55c78167c3e0 - std::panicking::default_hook::{{closure}}::h2dde766cd83a333a
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:210:50
   9:     0x55c78167bfd2 - std::panicking::default_hook::h501e3b2e134eb149
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:224:9
  10:     0x55c78167cbd4 - std::panicking::rust_panic_with_hook::hc09e869c4cf00885
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:624:17
  11:     0x55c78167c6b0 - std::panicking::begin_panic_handler::{{closure}}::hc2c6d70142458fc8
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:521:13
  12:     0x55c781679324 - std::sys_common::backtrace::__rust_end_short_backtrace::had58f7c459a1cd6e
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x55c78167c619 - rust_begin_unwind
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:517:5
  14:     0x55c770c62f61 - core::panicking::panic_fmt::hf443e5eeb6cc9eab
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panicking.rs:96:14
  15:     0x55c7816d298e - core::panicking::assert_failed_inner::hbc9a6c95e89a44e1
  16:     0x55c774cc1c3c - core::panicking::assert_failed::hc50de6d897a8d9ab
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panicking.rs:134:5
  17:     0x55c7716afc9a - test::assert_test_result::h172e063bb6dccfb9
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:194:5
  18:     0x55c77134b544 - databend_query::servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table::{{closure}}::h7d996a3870dca1a3
                               at /home/runner/work/databend/databend/query/src/servers/clickhouse/clickhouse_handler_test.rs:71:7
  19:     0x55c77134b544 - core::ops::function::FnOnce::call_once::h4d46c576d5961340
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/ops/function.rs:227:5
  20:     0x55c773029723 - core::ops::function::FnOnce::call_once::habda61fc7f1d22a8
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/ops/function.rs:227:5
  21:     0x55c773029723 - test::__rust_begin_short_backtrace::h5821bcd7faedd4ba
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:578:5
  22:     0x55c773028188 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h1a054a500b1f5f44
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
  23:     0x55c773028188 - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h9bf751b83fef5d31
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
  24:     0x55c773028188 - std::panicking::try::do_call::h17dfa45285c9488d
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
  25:     0x55c773028188 - std::panicking::try::h8b3678190e724618
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
  26:     0x55c773028188 - std::panic::catch_unwind::h3fe7822b5c965cd0
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
  27:     0x55c773028188 - test::run_test_in_process::hc302ac0a3797494c
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:601:18
  28:     0x55c773028188 - test::run_test::run_test_inner::{{closure}}::h6ba05affdd12aa23
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:493:39
  29:     0x55c772ff2efe - test::run_test::run_test_inner::{{closure}}::h97b084abc2d4f737
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/test/src/lib.rs:520:37
  30:     0x55c772ff2efe - std::sys_common::backtrace::__rust_begin_short_backtrace::h78f083d2b9ff0fdb
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys_common/backtrace.rs:125:18
  31:     0x55c772ff8648 - std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}::h7615918d3bd30972
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/mod.rs:481:17
  32:     0x55c772ff8648 - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::hcf655186d919dba6
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/panic/unwind_safe.rs:271:9
  33:     0x55c772ff8648 - std::panicking::try::do_call::h0a964b39e652f54e
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:403:40
  34:     0x55c772ff8648 - std::panicking::try::hebf5b3e02ddfd6d2
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panicking.rs:367:19
  35:     0x55c772ff8648 - std::panic::catch_unwind::hb2c7a7e68edb9779
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/panic.rs:129:14
  36:     0x55c772ff8648 - std::thread::Builder::spawn_unchecked::{{closure}}::h2abe6406bce13b45
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/thread/mod.rs:480:30
  37:     0x55c772ff8648 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h10f497d00829d905
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/ops/function.rs:227:5
  38:     0x55c781689243 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::h59eef3b9c8a82350
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
  39:     0x55c781689243 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hb5bbe017c347469c
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/alloc/src/boxed.rs:1636:9
  40:     0x55c781689243 - std::sys::unix::thread::Thread::new::thread_start::h62931528f61e35f5
                               at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/std/src/sys/unix/thread.rs:106:17
  41:     0x7f8a3294a609 - start_thread
  42:     0x7f8a326f4293 - clone
  43:                0x0 - <unknown>


failures:
    servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table
```

**Comment by BohuTANG** [comment]
cc @dantengsky @drmingdrmer

**Comment by dantengsky** [comment]
in this case, there is another version of table "default::T1"  inside our embedded meta, two ways come into mind to resolve this issue:
  - use a unique `meta_embedded_dir` for this test case
  - use a unique table name for this test case

@drmingdrmer I am not sure which one is more proper, any suggestions?

@sundy-li, I have noticed that there is an error message

> **test servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table has been running for over 60 seconds**
> test servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table ... FAILED

is it supposed to appear in this case?


A digest error stack:

```
test servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table has been running for over 60 seconds
test servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table ... FAILED

failures:

---- servers::clickhouse::clickhouse_handler_test::test_clickhouse_insert_to_fuse_table stdout ----
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: Code: 2407, displayText = targeting version 1, current version 2.

   0: common_exception::exception::ErrorCode::TableVersionMissMatch
             at /home/runner/work/databend/databend/common/exception/src/exception.rs:110:77
   1: common_meta_embedded::meta_api_impl::<impl common_meta_api::meta_api::MetaApi for common_meta_embedded::meta_embedded::MetaEmbedded>::upsert_table_option::{{closure}}
             at /home/runner/work/databend/databend/common/meta/embedded/src/meta_api_impl.rs:266:24
......

Error: Code: 1000, displayText = Error insert table: Driver(Timeout).

   0: common_exception::exception::ErrorCode::UnknownException
             at /home/runner/work/databend/databend/common/exception/src/exception.rs:110:77
   1: databend_query::servers::clickhouse::clickhouse_handler_test::insert::{{closure}}
```

**Comment by sundy-li** [comment]
>  is it supposed to appear in this case?

No, let's resolve the `TableVersionMissMatch` firstly.

**Comment by dantengsky** [comment]
> No, let's resolve the TableVersionMissMatch firstly.

roger

**Comment by drmingdrmer** [comment]
> * use a unique `meta_embedded_dir` for this test case
> * use a unique table name for this test case

I'm not quite sure whether it is the root cause:
In the latest branch `main`, embedded meta-store always is created in a temp dir and uses a unique sled-tree name.
Thus unit tests should not affect each other.

https://github.com/datafuselabs/databend/blob/969abfece84a3b4050d8a96262700606760a31f0/query/src/catalogs/impls/catalog/metastore_catalog.rs#L81
https://github.com/datafuselabs/databend/blob/969abfece84a3b4050d8a96262700606760a31f0/query/src/catalogs/backends/impls/embedded_backend.rs#L31-L40
https://github.com/datafuselabs/databend/blob/969abfece84a3b4050d8a96262700606760a31f0/common/meta/embedded/src/meta_embedded.rs#L67-L80


My guess is:
`MetaCached` caches the table info by table-id. Thus some CAS-loop can not re-fetch the latest state.

Not very sure what the cache is for.
@dantengsky added this cache for some reason and may help on reconsidering the design.

https://github.com/datafuselabs/databend/blob/969abfece84a3b4050d8a96262700606760a31f0/query/src/catalogs/backends/impls/meta_cached.rs#L89-L104

**Comment by dantengsky** [comment]
> I'm not quite sure whether it is the root cause:
> In the latest branch main, embedded meta-store always is created in a temp dir and uses a unique sled-tree name.
> ....

roger & thanks

>Not very sure what the cache is for.
> @dantengsky added this cache for some reason and may help on reconsidering the design.

it used to be,  if recall correctly, a cache for versioned table meta info. since we no longer allow accessing table by id and version (I am ok with it), I think we can remove this cache

**Comment by BohuTANG** [comment]
This issue stille exists(triggered by the main branch CI):
https://github.com/datafuselabs/databend/runs/4077452057?check_suite_focus=true

**Comment by drmingdrmer** [comment]
Another consistency issue with `MetaCache`(now cache has been removed from `main` branch):

Two concurrent `get_table_by_id()` has chance overrides each others' data:

```rust
    async fn get_table_by_id(&self, table_id: MetaId) -> Result<(TableIdent, Arc<TableMeta>)> {
        {
            let mut cached = self.table_meta_cache.write().await;
            if let Some(meta) = cached.get(&(table_id, 0)) {
                return Ok(meta.clone());
            }
        }

        let reply = self.inner.get_table_by_id(table_id).await?;
        let ident = reply.0.clone();

        let mut cache = self.table_meta_cache.write().await;
        // TODO version
        cache.put((ident.table_id, 0), reply.clone());
        Ok(reply)
    }
```

```
A get() no-cache  meta.get() -> id=1,ver=1                                            write-cache(id=1,ver=1)
B get() no-cache  meta.get() -> id=2,ver=2 write-cache(id=1,ver=2) 
--------------------------------------------------------------------------------------------> time
```

I do not in the test case `test_clickhouse_insert_to_fuse_table` if possible there are 2 concurrent `get_table_by_id()` is running. 🤔

**Comment by dantengsky** [comment]
> I do not know in the test case test_clickhouse_insert_to_fuse_table if possible there are 2 concurrent get_table_by_id() is running. 

for commit 44c21ebe4  (which triggered error `TableVersionMissMatch` in CI) ,  `MetaCache::get_table_by_id` is not invoked  by `test_clickhouse_insert_to_fuse_table`

**Comment by drmingdrmer** [comment]
### Maybe it is because DatabendQueryContextShared cached table info?
@zhang2014 is it possible to happen? I'm not very familiar with session and context, need your help XDDD.

I added several println to `DatabendQueryContextShared::get_table()`:

https://github.com/drmingdrmer/databend/blob/ecc7cc9b80ac2b12f195b59fd2d6f7451dddc1a6/query/src/sessions/context_shared.rs#L117-L148

```rust
    pub fn get_table(&self, database: &str, table: &str) -> Result<Arc<dyn Table>> {
        println!("ContextShared::get_table: {} {}", database, table);
        // Always get same table metadata in the same query
        let table_meta_key = (database.to_string(), table.to_string());

        let mut tables_refs = self.tables_refs.lock();

        let ent = match tables_refs.entry(table_meta_key) {
            Entry::Occupied(entry) => {
                println!(
                    "ContextShared::Occupied: {} {}, {:?}",
                    database,
                    table,
                    entry.get().get_table_info()
                );
                entry.get().clone()
            }
            Entry::Vacant(entry) => {
                let catalog = self.get_catalog();
                let rt = self.try_get_runtime()?;
                let database = database.to_string();
                let table = table.to_string();
                println!("ContextShared::Vacant: {} {}", database, table);
                let t = (async move { catalog.get_table(&database, &table).await })
                    .wait_in(&rt, Some(Duration::from_millis(5000)))??;

                entry.insert(t).clone()
            }
        };

        Ok(ent)
    }
```

And when panicked, the output is as:
https://github.com/datafuselabs/databend/runs/4088540597?check_suite_focus=true
 
```
ContextShared::get_table: default tbl_50fbe102595f446aafe6a7ba934e74dc
ContextShared::Vacant: default tbl_50fbe102595f446aafe6a7ba934e74dc
MetaStoreCatalog::get_table: default tbl_50fbe102595f446aafe6a7ba934e74dc
MetaStoreCatalog::get_table: default tbl_50fbe102595f446aafe6a7ba934e74dc
get_table: default tbl_50fbe102595f446aafe6a7ba934e74dc
got table: TableInfo { ident: TableIdent { table_id: 1, version: 1 }, desc: "'default'.'tbl_50fbe102595f446aafe6a7ba934e74dc'", name: "tbl_50fbe102595f446aafe6a7ba934e74dc", meta: TableMeta { schema: DataSchema { fields: [DataField { name: "a", data_type: UInt32, nullable: false }, DataField { name: "b", data_type: UInt64, nullable: false }, DataField { name: "c", data_type: String, nullable: false }], metadata: {} }, engine: "fuse", options: {} } }

ContextShared::get_table: default tbl_50fbe102595f446aafe6a7ba934e74dc
ContextShared::Occupied: default tbl_50fbe102595f446aafe6a7ba934e74dc, TableInfo { ident: TableIdent { table_id: 1, version: 1 }, desc: "'default'.'tbl_50fbe102595f446aafe6a7ba934e74dc'", name: "tbl_50fbe102595f446aafe6a7ba934e74dc", meta: TableMeta { schema: DataSchema { fields: [DataField { name: "a", data_type: UInt32, nullable: false }, DataField { name: "b", data_type: UInt64, nullable: false }, DataField { name: "c", data_type: String, nullable: false }], metadata: {} }, engine: "fuse", options: {} } }
upsert_table_option: 1 1 SNAPSHOT_LOC _ss/b0d0c56e54834b2a925471ca32b93473

ContextShared::get_table: default tbl_50fbe102595f446aafe6a7ba934e74dc
ContextShared::Occupied: default tbl_50fbe102595f446aafe6a7ba934e74dc, TableInfo { ident: TableIdent { table_id: 1, version: 1 }, desc: "'default'.'tbl_50fbe102595f446aafe6a7ba934e74dc'", name: "tbl_50fbe102595f446aafe6a7ba934e74dc", meta: TableMeta { schema: DataSchema { fields: [DataField { name: "a", data_type: UInt32, nullable: false }, DataField { name: "b", data_type: UInt64, nullable: false }, DataField { name: "c", data_type: String, nullable: false }], metadata: {} }, engine: "fuse", options: {} } }

ContextShared::get_table: default tbl_50fbe102595f446aafe6a7ba934e74dc
ContextShared::Occupied: default tbl_50fbe102595f446aafe6a7ba934e74dc, TableInfo { ident: TableIdent { table_id: 1, version: 1 }, desc: "'default'.'tbl_50fbe102595f446aafe6a7ba934e74dc'", name: "tbl_50fbe102595f446aafe6a7ba934e74dc", meta: TableMeta { schema: DataSchema { fields: [DataField { name: "a", data_type: UInt32, nullable: false }, DataField { name: "b", data_type: UInt64, nullable: false }, DataField { name: "c", data_type: String, nullable: false }], metadata: {} }, engine: "fuse", options: {} } }
upsert_table_option: 1 1 SNAPSHOT_LOC _ss/f905a5b3305b41f083a70e9801a8bed3
```

It shows that every time it reads the table from cache thus the caller can not get the latest table info(ver=2).

And a `DatabendQueryContextShared` is created from the a cached one in the`Session.mutable_state`:

https://github.com/datafuselabs/databend/blob/50e6ec1b8b5301a69e19c33b08fe204468cd221d/query/src/sessions/session.rs#L127-L154

**Comment by dantengsky** [comment]
@drmingdrmer FYI, I have done a similar debug:


```rust
    pub fn get_table(&self, database: &str, table: &str) -> Result<Arc<dyn Table>> {
        // Always get same table metadata in the same query
        let table_meta_key = (database.to_string(), table.to_string());

        let mut tables_refs = self.tables_refs.lock();
        eprintln!(
            "context shared getting table : cache ptr {:?}, session id {}, t {:?}",
            std::ptr::addr_of!(self.tables_refs),
            self.session.id,
            std::thread::current()
        );
```

```
UT, insert1 | B
context shared getting table : cache ptr 0x7fe0c42593b0, session id 616e0fc3-a33b-4e88-807c-d02447844b55, t Thread { id: ThreadId(28), name: Some("tokio-runtime-worker"), .. }
context shared getting table : cache ptr 0x7fe0c42593b0, session id 616e0fc3-a33b-4e88-807c-d02447844b55, t Thread { id: ThreadId(51), name: Some("tokio-runtime-worker"), .. }
appending data , self version 1, t Thread { id: ThreadId(51), name: Some("tokio-runtime-worker"), .. }
UT, insert1 | E
UT, insert2 | B
context shared getting table : cache ptr 0x7fe0c4259fb0, session id 616e0fc3-a33b-4e88-807c-d02447844b55, t Thread { id: ThreadId(28), name: Some("tokio-runtime-worker"), .. }
context shared getting table : cache ptr 0x7fe0c4259fb0, session id 616e0fc3-a33b-4e88-807c-d02447844b55, t Thread { id: ThreadId(79), name: Some("tokio-runtime-worker"), .. }
appending data , self version 2, t Thread { id: ThreadId(79), name: Some("tokio-runtime-worker"), .. }
UT, insert2 | E
```

- the two insertions belongs to the same session
- cache **seems** to be different

**Comment by drmingdrmer** [comment]
@dantengsky 
It is weird 🤔 
- The `cache ptr`(tables_refs) are the same.
- `get_table()` seems to be the only place where `tables_refs` is used.

Thus `get_table()` should always return the same result(ver)

**Comment by dantengsky** [comment]
same feeling…

**Comment by zhang2014** [comment]
/assignme 

We can remove context_shared from session after #2573

**Comment by drmingdrmer** [comment]
Speculation from @zhang2014  on this issue:
- Shared-context(the container of table cache) is removed from `session` once the last `context` is dropped.
- If a `context` is somehow kept by an unfinished `future`, the `shared-context` will stay there in `session`.
- The next query may have chance reusing the previous one.

**Comment by zhang2014** [comment]
It hasn't been fixed yet.

**Comment by zhang2014** [comment]
fixed by https://github.com/datafuselabs/databend/pull/4845
